source/plugin/commands/argumentparsing.rst

```diff
@@ -0,0 +1,176 @@
+      - mutiple ``ServerPlayer`` instances
+    * - ``user``
+      - Require the argument to be a user. *May return multiple users!*
+      - multiple ``User`` instances
+    * - ``world``
+      - Require the argument to be a world (only loaded worlds)
+      - multiple ``ServerWorld`` instances
+    * - ``location``
+      - Require the argument to be a location
+      - one ``ServerLocation``
+    * - ``vector3d``
+      - Require the argument to be a vector
+      - one ``Vector3d``
+    * - ``rotation``
+      - Require the argument to be a vector, but returns the senders rotation if no value was specified
+      - one ``Vector3d``
+    * - ``registryElement``
+      - Require the argument to be the resoure key of one of the specified elements
+      - multiple matching elements of the specified registry type
+    * - ``choices``
+      - Returns an argument that allows selecting from a limited set of values
+      - one specified value
+    * - ``literal``
+      - Require the argument to match one of the specified literals
+      - one specified literal
+    * - ``enumValue``
+      - Require the argument to be a enum
+      - one ``Enum`` 
+    * - ``seq``
+      - Builds a sequence of commands
+      - inherited
+
+.. tip::
+    See the Javadocs for :javadoc:`Parameter` for more information
+
+.. warning::
+
+    Don't expect that a ``Parameter`` will only ever return a single value, 
+    a lot of them support multiple return values; some might support regular expressions or use command selector. 
+    This is intensional as it makes commands easier to use. Example ``/tell @a BanditPlayer has cookies!``.
+    
+Custom Parameter
+~~~~~~~~~~~~~~~~
+
+It is possible to create custom command elements. Example Vector2i. This is done though the 
+:javadoc:`Parameter#builder(Class)` method, which returns a :javadoc:`Parameter.Builder` where all data of the parameter
+is needed to be provided. Once done call the ``build`` method to build the parameter.
+
+When building a new parameter, only the ``parser`` and ``key`` are required for build. The ``parser`` is where the logic
+of mapping the ``String`` input to the desired value is.
+
+Example: Creating a Custom Parameter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example we will make a Vector2i parameter. This requires reading two Integers and converting them to a Vector2i.
+
+.. code-block:: java
+
+    Parameter.Value<Vector2i> vectorParameter = Parameter
+      .builder(Vector2i.class)
+      .addParser((
+        Parameter.Key<Vector2i> parameterKey, 
+        ArgumentReader.Mutable reader, 
+        CommandContext.Builder context) -> {
+          int x = reader.parseInt();
+          int y = reader.parseInt();
+          return new Vector2i(x, y);
+      })
+      .key("vector")
+      .build();
+
+.. tip::
+
+    When building a new parameter, you can base your parameter from a exsiting parameter. Example: taking 
```

Comment:
"you can base your parameter on an existing parameter".
i.e. from -> on 
and fix typo

Reply:
Updated